packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -207,7 +207,11 @@ rocksdb::Status ArbCore::initialize(const LoadedExecutable& executable) {
         }
 
         std::vector<unsigned char> value_data;
-        marshal_uint256_t(machine->hash(), value_data);
+        auto machine_hash = machine->hash();
+        if (!machine_hash) {
+            std::cerr << "failed to compute initial machine hash" << std::endl;
```

Comment:
This should return or throw since otherwise we'll be dereferencing an empty optional after this

Reply:
👍